### PR TITLE
Feature/imageable-model

### PR DESCRIPTION
**Pull request by Jifflis**
# 📲 What #
- Create Imageable model
 
# 🛠 How #
- added Imageable model class and  Imageable migration class
 
# ✅ Acceptance criteria #

# ⏰ TODO #
None

# 🔗 References #
https://trello.com/c/nl4MiP6M/5-wallet-model

### CODE REVIEW

**Review by jaimejazarenoiii — CHANGES_REQUESTED** [review]
Please follow our PR template @Jifflis  thanks

**Review by jaimejazarenoiii — COMMENTED** [review]
Did you run rubocop here? I'm pretty sure there are errors should prompt here. There's using double quotation and else

**Review by jaimejazarenoiii — COMMENTED** [review]
It's nice that u understand how to handle models very good. but https://guides.rubyonrails.org/association_basics.html#polymorphic-associations this is the way for polymorphic models. Thank you

**Comment by jaimejazarenoiii** [comment]
I created an overcommit hook so that rubocop(linter) runs before user can even commit

File: app/models/imageable.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+#  type           :integer
+#  created_at     :datetime         not null
+#  updated_at     :datetime         not null
+#  imageable_id   :bigint           not null
+#
+# Indexes
+#
+#  index_imageables_on_imageable  (imageable_type,imageable_id)
+#
+class Imageable < ApplicationRecord
```
Comment (jaimejazarenoiii) [inline app/models/imageable.rb]: I think the class should be just Image? Then the polymorphic association is the imageable one

File: app/models/imageable.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  validates :image, presence: true, uniqueness: true
+  enum type: %i[type1 type2 type3]
```
Comment (jaimejazarenoiii) [inline app/models/imageable.rb]: Maybe for now we set two types? Icons and Photo

**Review by jaimejazarenoiii — CHANGES_REQUESTED** [review]

File: app/models/imageable.rb (hunk 1 above)
Comment (Jifflis) [reply app/models/imageable.rb]: Much better!

File: app/models/imageable.rb (hunk 2 above)
Comment (Jifflis) [reply app/models/imageable.rb]: Alright!

File: app/models/imageable.rb (hunk 1 above)
Comment (jaimejazarenoiii) [reply app/models/imageable.rb]: Forgot about this https://edgeguides.rubyonrails.org/active_storage_overview.html

**Review by rexendy — APPROVED** [review]

**Pull request merged**